aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -94,32 +97,88 @@ public Orchestrator(
-                subscriptionProcessor.startDrainingMutationBuffer();
-                LOG.info("Cloud synchronization is now fully active.");
-            }));
+        // Only start if it's stopped.
+        if (OrchestratorStatus.STOPPED.equals(status.get())) {
+            LOG.debug("Starting the orchestrator.");
+            status.compareAndSet(OrchestratorStatus.STOPPED, OrchestratorStatus.STARTING);
+            initializationCompletable = mutationOutbox.load()
+                .andThen(
+                    Completable.fromAction(() -> {
+                        if (!storageObserver.isObservingStorageChanges()) {
+                            LOG.debug("Starting local storage observer.");
+                            storageObserver.startObservingStorageChanges();
+                        }
+                        if (!subscriptionProcessor.isObservingSubscriptionEvents()) {
+                            LOG.debug("Starting subscription processor.");
+                            subscriptionProcessor.startSubscriptions();
+                        }
+                        syncProcessor.hydrate().blockingAwait();
+                        if (!mutationProcessor.isDrainingMutationOutbox()) {
+                            LOG.debug("Starting mutation processor.");
+                            mutationProcessor.startDrainingMutationOutbox();
+                        }
+                        if (!subscriptionProcessor.isDrainingMutationBuffer()) {
+                            LOG.debug("Starting draining mutation buffer.");
+                            subscriptionProcessor.startDrainingMutationBuffer();
+                        }
+                        status.compareAndSet(OrchestratorStatus.STARTING, OrchestratorStatus.STARTED);
+                    })
+                );
+        }
+        return initializationCompletable;
     }
 
     /**
      * Stop all model synchronization.
      */
     public void stop() {
-        LOG.info("Intentionally stopping cloud synchronization, now.");
-        storageObserver.stopObservingStorageChanges();
-        subscriptionProcessor.stopAllSubscriptionActivity();
-        mutationProcessor.stopDrainingMutationOutbox();
+        if (OrchestratorStatus.STARTED.equals(status.get())) {
+            LOG.info("Intentionally stopping cloud synchronization, now.");
+            status.compareAndSet(OrchestratorStatus.STARTED, OrchestratorStatus.STOPPING);
+            subscriptionProcessor.stopAllSubscriptionActivity();
+            storageObserver.stopObservingStorageChanges();
+            mutationProcessor.stopDrainingMutationOutbox();
+            status.compareAndSet(OrchestratorStatus.STOPPING, OrchestratorStatus.STOPPED);
+            LOG.debug("Stopped remote synchronization.");
+        }
+
+    }
+
+    /**
+     * Represents possible status of the orchestrator.
+     */
+    enum OrchestratorStatus {
+        /**
+         * The orchestrator is stopping.
```

Comment:
True 'nuff, but when it is stopping, what does that mean for the system -- request to `start()` will be denied? What are the valid input and output states, from this step?

Same Q for all

Reply:
I went through and updated the comments for each of those to be a bit more descriptive.